### PR TITLE
`Flow` can use `Base Flow`

### PR DESCRIPTION
**Pull request by PerisicJere**

### CODE REVIEW

**Review by vinkobuble — CHANGES_REQUESTED** [review]

File: library/src/fy_library/flows/imports/flow_imports_fy.py [hunk 1]
```diff
@@ -81,20 +98,28 @@ def __init__(
...
         self._parsed_fy_py_files_map_by_key = parsed_fy_py_files_map_by_key
         self._mixin_import_map = mixin_import_map
+        self._parsed_fy_py_file = parsed_fy_py_file
         super().__init__(*args, **kwargs)
 
     def __call__(self) -> List[str]:
         # fy:end <<<===
+        correct_base_flow_import = (
+            self._import_flow_base
+            if not self._import_base_flow
```
[the remark this answers is not in the document]
Comment (PerisicJere) [reply library/src/fy_library/flows/imports/flow_imports_fy.py:115]: Implemented `if self._import_base_flow == []`

File: library/src/fy_library/mixins/property/parsed_flow_fy_py_file/using_parsed_fy_py_file_fy.py [hunk 2]
```diff
@@ -70,6 +70,9 @@ def _parsed_flow_fy_py_file(self) -> ParsedFlowFyPyFile:
             file_path=self._fy_py_file_to_parse,
             user_imports=self._flow_file_split.user_imports,
             flow_name=flow_name,
+            declared_base_flow=self._flow_file_split.declared_base_flow
+            if self._flow_file_split.declared_base_flow
```
[the remark this answers is not in the document]
Comment (PerisicJere) [reply library/src/fy_library/mixins/property/parsed_flow_fy_py_file/using_parsed_fy_py_file_fy.py:74]: Added `!= ""`

File: library/src/fy_library/mixins/property/imports/import_base_flow__using_declared_base_flow_name_fy.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+):
+    @cached_property
+    def _import_base_flow(self) -> List[str]:
+        # fy:end <<<===
+        correct_base_flow_import: List[str] = []
+        if self._base_flow_import != "":
+            correct_base_flow_import = [
+                self._mixin_import_map[self._base_flow_import.lower()]
+            ]
+        return correct_base_flow_import
```
[the remark this answers is not in the document]
Comment (PerisicJere) [reply library/src/fy_library/mixins/property/imports/import_base_flow__using_declared_base_flow_name_fy.py:42]:
```
       if self._base_flow_import == "":
            return []
        return [
                self._mixin_import_map[self._base_flow_import.lower()]
            ]
  ```

File: library/src/fy_library/mixins/property/flow_file_split/using_flow_regex_fy.py [hunk 4]
```diff
@@ -21,7 +21,8 @@
 )
 
 _FLOW_STRING_SPLIT_REGEX: Final = re.compile(
-    rf"flow\s+(?P<flow_name>{FY_ENTITY_REGEX_STRING})\s+->"
+    rf"flow\s+(?P<flow_name>{FY_ENTITY_REGEX_STRING})\s*"
+    rf"(?P<declared_base_flow>\(({FY_ENTITY_REGEX_STRING})\))?\s+->"
```
[the remark this answers is not in the document]
Comment (PerisicJere) [reply library/src/fy_library/mixins/property/flow_file_split/using_flow_regex_fy.py:25]: `rf"\(?(?P<declared_base_flow>{FY_ENTITY_REGEX_STRING})?\)?\s+->"`
Fixed it with this regex, always returns a list of 5 elements.

File: library/src/fy_library/mixins/property/flow_file_split/using_flow_regex_fy.py [hunk 5]
```diff
@@ -37,13 +38,20 @@ def _flow_file_split(self) -> FlowFileSplitModel:
         # fy:end <<<===
         flow_file_split = _FLOW_STRING_SPLIT_REGEX.split(self._fy_code)
 
+        declared_base_flow: str = ""
+        if len(flow_file_split) == 6:
+            declared_base_flow = flow_file_split[3]
+            flow_file_split.pop(3)
+            flow_file_split.pop(2)
+
         assert (
```
[the remark this answers is not in the document]
Comment (PerisicJere) [reply library/src/fy_library/mixins/property/flow_file_split/using_flow_regex_fy.py:41]: After the change to the regex, there was no need for that.

**Review by vinkobuble — APPROVED** [review]

**Pull request merged**